### DISCUSSION TITLE
GTK4 apps don't respect dark theme in i3

### DISCUSSION THREAD

**Discussion by artichoked1** [i3/i3#5896] [Q&A]
Hi, this might be a bit of an obscure issue and could be more related to gnome rather than i3 so apologies in advance if this might not be the best place to put it.

i3 v4.22
Ubuntu 23.10
GNOME 45

Ive been a long-time GNOME user but recently, I wanted to try a tiling window manager so I have both GNOME and i3 co-existing on my system. When I log in after a fresh boot into GNOME, all my GTK3/4 apps are in dark mode as they should be. When I logout and log back in but this time using i3, GTK 4 apps like nautilus, gnome-clocks, gnome-calendar are in light mode.
Wierdly, if I log out again and back into gnome, the GTK apps are still in lightmode there too, regardless of whether I toggle the dark mode slider on or off. It is only if I do a full reboot that they are in dark mode again.

I have tried the following solutions:

- Making sure in the `settings.ini` file in both `~/.config/gtk-4.0/`  and `~/.config/gtk-3.0/`  have `application-gtk-prefer-dark-theme` set to `1`

- Making sure the following are set through dconf-editor:

```
/org/gnome/desktop/interface/gtk-theme = 'Yaru-dark'
/org/gnome/desktop/interface/gtk-key-theme = Default
/org/gnome/desktop/interface/color-scheme = Default or 'prefer-dark'
```

- Starting a settings daemon in my i3 config like `gsd-xsettings` or `xsettingsd` to apply my current GTK settings.

**Comment by orestisfl** [comment]
Stab in the dark here, but have you tried
```
gsettings set org.gnome.desktop.interface color-scheme prefer-dark
````
?

Maybe give https://wiki.archlinux.org/title/Dark_mode_switching and https://wiki.archlinux.org/title/GTK#Dark_theme_variant a good read.

**Reply by artichoked1** [reply]
Yes I have tried that but still the same result. I like the pun.

**Comment by jthistle** [comment]
I have the same issue, have not been able to fix, even with `~/.config/gtk-4.0/settings.ini` as

```ini
[Settings]
gtk-application-prefer-dark-theme=1
```

I note that I get an error with Nautilus, which is

```
(org.gnome.Nautilus:250374): Adwaita-WARNING **: 11:33:48.737: Using GtkSettings:gtk-application-prefer-dark-theme with libadwaita is unsupported. Please use AdwStyleManager:color-scheme instead.
```

which is weird, because `gnome-tweaks` says I'm using Yaru, not Adwaita.

**Comment by lexxxel** [comment]
have you checked the solution from https://unix.stackexchange.com/questions/761808/how-to-globally-set-dark-mode-for-gtk-4-apps
TL;DR: `export GTK_THEME=Adwaita-dark` and test your application. For me it finally helped with Nautilus.

**Reply by xdevs23** [reply]
This seems to apply the wrong theme. Buttons look differently. in GTK4 (libadwaita) they are flatter. Overall, with Adwaita-dark it looks weird.

**Comment by jthistle** [comment]
I fixed this on 23.10 by creating `$HOME/.config/xdg-desktop-portal/portals.conf` with contents: 

```conf
[preferred]
default=gtk;wlr
```

After a restart it was fixed for me. See also: https://discourse.gnome.org/t/using-adwaita-dark-on-another-compositor/17563/10

**Reply by guilhermetk** [reply]
That worked for me after upgrading from Ubuntu 22.04 to 24.04. Thanks!

**Reply by squeaktoy** [reply]
`gtk` doesn't have to be upfront. It just needs to be present in the list.

**Reply by peterrus** [reply]
Confirming this also works on the wayland session! (https://github.com/orgs/regolith-linux/discussions/1044#discussioncomment-10691886)

**Reply by sergii-dudar** [reply]
I have spend and tried a lot options without much success, before I have found your answer why my themes not appying gkt apps on my i3, qtile setups, thansk you very much, you my hero!

**Reply by Abdelaziz-LAISSAOUI** [reply]
thanks a lot!

**Reply by robin-thoni** [reply]
Using Regolith on Ubuntu 24 on X11, I had to remove the `wlr` part, or apps would take 1 or 2 minutes to start, waiting for `xdg-desktop-portal-wlr` to timeout (AFAIU, wlr is a Wayland compositor, thus not applicable on X11)

**Reply by SayantanRC** [reply]
This worked in my arch install (with cinnamon)

**Comment by Aaditya18SP** [comment]
I recently faced this issue in GTK2 and 3 apps where they wouldn't respect the dark theme, but didnt have any issue with GTK4 apps.
I fixed the issue for GTK2 and 3 apps by editing the theme from `gnome-tweaks`. 
Just give it a shot, switch the theme to `Adwaita-dark` from gnome-tweaks while in `Gnome` and then reboot/log out and check in i3.

**Comment by unt3rhofer** [comment]
This is what worked for me:
go to 
~/.config/gtk-4.0/settings.ini
(create both the directory and the file if they are missing)
then open settings.ini in your preffered text editor (vim or nano for example)
and put this inside
[Settings]
gtk-theme-name=YourThemeName
gtk-icon-theme-name=YourIconThemeName

mine looks like this:
[Settings]
gtk-theme-name=Arc-Dark
gtk-icon-theme-name=Reversal-grey-dark

You can choose any theme you like

**Comment by maynouf** [comment]
I had the same problem with Linux Mint 22 with Cinnamon. Keep in mind that without a GNOME/XFCE session, GTK apps have nothing to read theme settings from. 

`xsettingsd` is a lightweight daemon that fills that role:

```sudo apt install xsettingsd```

Then you can create `~/.config/xsettingsd/xsettingsd.conf` with:
```
Net/ThemeName "Adwaita-dark"
Net/IconThemeName "Adwaita"
```
(keep in mind that you've got to have those themes installed)

Check if it worked with:

`xsettingsd &`

To make it permanent, add `exec --no-startup-id xsettingsd` to your i3 config. 

If you use regolith you can make a file with this command in `~/.config/regolith3/i3/config.d/xsettingsd.conf`. (I mention it because it was my case, and someone mind find that helpful)
